MapboxSceneKit/Routes/TerrainNodeExtension+Routes.swift

```diff
@@ -81,6 +81,66 @@ extension TerrainNode {
             let position = self.positionForLocation(coord)
             scenePositions.append(position)
         }
+        scenePositions = subdivideAndSnapToTerrain(positions: scenePositions)
         return scenePositions
     }
+    
+    /// Resamples the provided route based on zoom level to avoid polylines that intersect or float above terrain.
+    ///
+    /// - Parameter positions: the positions to be re-sampled
+    /// - Returns: the final route, snapped to the terrainNode's surface
+    fileprivate func subdivideAndSnapToTerrain(positions: [SCNVector3]) -> [SCNVector3] {
+        
+        guard let firstPosition = positions.first else { return [] }
+        var newPositions =  [SCNVector3]()
+        newPositions.append(firstPosition)
+        
+        //re-use a single bezierspline instance to interpolate each segment
+        var positionBezier: BezierSpline3D
+        
+        //for each segment...
+        for index in 1..<positions.count {
+            
+            let fromPostion = positions[index - 1]
+            let toPositon = positions[index]
+            
+            /*
+             Define the maximum meaningful elevation samples supported by the terrainnode's pixels per meter.
+             More samples would read the same pixel multiple times per line, creating jagged steps.
+             */
+            let lengthInMeters = Double((toPositon - fromPostion).length())
+            let maxSamplesInSegment = lengthInMeters / self.metersPerX
+            
+            //resample at a 1/5 of the maximum terrain resolution, to save on performance.
```

Comment:
Any special reason for 1/5? Just found that it worked well? (curious)

Reply:
Yes, it just seemed to work well across zoom levels. I think we need some kind of 'constants' class to store settings like this that you might want to change, but really shouldn't be defined with every function call.